libnavigation-core/src/main/java/com/mapbox/navigation/core/accounts/BillingController.kt

```diff
@@ -382,31 +386,32 @@ internal class BillingController(
      * ignoring origin.
      */
     private fun getRemainingWaypointsOnRoute(routeProgress: RouteProgress?): List<Point>? {
-        return routeProgress?.navigationRoute?.let { route ->
-            val waypoints = route.waypoints
-            waypoints?.drop(
-                (waypoints.size - routeProgress.remainingWaypoints).coerceAtLeast(1)
-            )?.map {
-                it.location()
-            }
+        if (routeProgress == null) return null
+        return getWaypointsOnRoute(routeProgress.navigationRoute) {
```

Comment:
Well, in the beginning `getWaypointsOnRoute` accepted nullable so I removed the let. Then it turned out it would be better to accept non-null. So I added the guard.
I find it a bit more readable and it keeps the code to the right (while with let block it moves to the left a bit). Absolutely no strong opinion here though.